image/bmp.ksy

```diff
@@ -52,53 +125,175 @@ types:
+      - id: header_size
+        type: u4
+      - id: header
+        -orig-id: bmciHeader
+        size: header_size - 4
+        type: bitmap_header(header_size)
+      - id: color_mask
+        type: color_mask(header.bitmap_info_ext.compression == compressions::alpha_bitfields)
+        if: not _io.eof and is_color_mask_here
+        doc: Valid only for BITMAPINFOHEADER, in all headers extending it the masks are contained in the header itself.
+      - id: color_table
+        -orig-id: bmciColors
+        size-eos: true
+        type: 'color_table(
+            not header.is_core_header,
+            header.extends_bitmap_info ? header.bitmap_info_ext.num_colors_used : 0
+          )'
+        if: not _io.eof
+    instances:
+      is_color_mask_here:
+        value: >-
+          header.header_size == header_type::bitmap_info_header.to_i
+            and (header.bitmap_info_ext.compression == compressions::bitfields or header.bitmap_info_ext.compression == compressions::alpha_bitfields)
+      is_color_mask_given:
+        value: >-
+          header.extends_bitmap_info
+          and (header.bitmap_info_ext.compression == compressions::bitfields or header.bitmap_info_ext.compression == compressions::alpha_bitfields)
+          and (is_color_mask_here or header.is_color_mask_here)
+      color_mask_given:
+        value: >-
+          is_color_mask_here
+            ? color_mask
+            : header.color_mask
+        if: is_color_mask_given
+      color_mask_red:
+        value: >-
+          is_color_mask_given
+            ? color_mask_given.red_mask
+            : header.bits_per_pixel == 16
+              ? 0b11111_00000_00000
+              : header.bits_per_pixel == 24 or header.bits_per_pixel == 32
+                ? 0xff_00_00
+                : 0
+        #         ^ uses fixed color palette, so color mask is N/A
+      color_mask_green:
+        value: >-
+          is_color_mask_given
+            ? color_mask_given.green_mask
+            : header.bits_per_pixel == 16
+              ? 0b00000_11111_00000
+              : header.bits_per_pixel == 24 or header.bits_per_pixel == 32
+                ? 0x00_ff_00
+                : 0
+      color_mask_blue:
+        value: >-
+          is_color_mask_given
+            ? color_mask_given.blue_mask
+            : header.bits_per_pixel == 16
+              ? 0b00000_00000_11111
+              : header.bits_per_pixel == 24 or header.bits_per_pixel == 32
+                ? 0x00_00_ff
+                : 0
+      color_mask_alpha:
+        value: >-
+          is_color_mask_given and color_mask_given.has_alpha_mask
+            ? color_mask_given.alpha_mask
+            : 0
+
+  bitmap_header:
+    -orig-id:
+      - BITMAPCOREHEADER
+      - OS21XBITMAPHEADER
+    doc-ref:
+      - https://docs.microsoft.com/cs-cz/windows/win32/api/wingdi/ns-wingdi-bitmapcoreheader
+      - https://www.fileformat.info/format/os2bmp/egff.htm#OS2BMP-DMYID.3.1
```

Comment:
Array in `doc-ref` is KS 0.9+ feature.

Reply:
OK, I haven't realized it. I thought about throwing the lists in `doc-ref` out to make it compatible with 0.8, but I anyway have to use `.as<f4>` (which is a 0.9 feature I suppose, doesn't work even in devel Web IDE) in `fixed_point_*` decimal types and I unfortunately  haven't figured out how to do it in 0.8. So I updated the `ks-version` to 0.9.

Reply:
Actually, this is another great point: we currently don't have a solid way to check these PRs before merging them. Implementing CI-style checks that will clearly show that this is not ok due to (following error messages) would be great...

Reply:
... and I completely agree, there's little point in clinging to 0.8. 0.9 is so much more advanced...